core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -221,6 +222,35 @@ public InternalEvent process(InternalEvent event) throws MuleException {
     initialiseIfInitialisable(pipeline);
   }
 
+  private ReactiveProcessor dispatchToFlow(Sink sink) {
+    if (source.getBackPressureStrategy() == WAIT) {
+      return publisher -> from(publisher).doOnNext(event -> sink.accept(event))
+          .flatMap(event -> Mono.from(event.getContext().getResponsePublisher()));
+    } else {
+      return publisher -> from(publisher).flatMap(event -> {
+        if (sink.emit(event)) {
+          return event.getContext().getResponsePublisher();
+        } else {
+          if (source.getBackPressureStrategy() == DROP) {
+            return empty();
+          } else {
+            RejectedExecutionException rejectedExecutionException = new RejectedExecutionException(format(OVERLOAD_ERROR_MESSAGE,
```

Comment:
maybe subclass `RejectedExecutionException` for this use?

Reply:
What value do you see in doing this?